lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -0,0 +1,760 @@
+    provider:
+        required: true
+        choices: [ 'selfsigned', 'assertonly', 'acme' ]
+        description:
+            - Name of the provider to use to generate/retrieve the OpenSSL certificate.
+              The 'assertonly' provider will not generate files and fail if the certificate file is missing.
+    force:
+        required: false
+        default: False
+        choices: [ True, False ]
+        description:
+            - Generate the certificate, even if it already exists.
+    csr_path:
+        required: false
+        description:
+            - Path to the Certificate Signing Request (CSR) used to generate this certificate. This is not required in 'assertonly' mode.
+
+    privatekey_path:
+        required: false
+        description:
+            - Path to the private key to use when signing the certificate.
+
+    privatekey_passphrase:
+        required: false
+        description:
+            - The passphrase for the privatekey.
+
+    selfsigned_digest:
+        required: false
+        default: "sha256"
+        description:
+            - Digest algorithm to be used when self-signing the certificate
+
+    selfsigned_notBefore:
+        required: false
+        description:
+            - The timestamp at which the certificate starts being valid. The timestamp is formatted as an ASN.1 TIME.
+
+    selfsigned_notAfter:
+        required: false
+        description:
+            - The timestamp at which the certificate stops being valid. The timestamp is formatted as an ASN.1 TIME.
+
+    acme_accountkey:
+        required: false
+        description:
+            - Path to the accountkey for the 'acme' provider
+
+    acme_challenge_path:
+        required: false
+        description:
+            - Path to the ACME challenge directory that is served on http://<HOST>:80/.well-known/acme-challenge/
+
+    signature_algorithms:
+        required: false
+        description:
+            - list of algorithms that you would accept the certificate to be signed with
+              (e.g. ['sha256WithRSAEncryption', 'sha512WithRSAEncryption']).
+
+    issuer:
+        required: false
+        description:
+            - Key/value pairs that must be present in the issuer name field of the certificate
+
+    subject:
+        required: false
+        description:
+            - Key/value pairs that must be present in the subject name field of the certificate
+
+    has_expired:
+        required: false
+        choices: [ True, False ]
+        description:
+            - Checks if the certificate is expired/not expired at the time the module is executed.
+
+    version:
+        required: false
+        description:
+            - Version of the certificate. Nowadays it should almost always be 3.
+
+    valid_at:
+        required: false
+        description:
+            - The certificate must be valid at this point in time. The timestamp is formatted as an ASN.1 TIME.
```

Comment:
https://www.obj-sys.com/asn1tutorial/node14.html states that appended time zones are also possible in ASN.1 format, so this might be a problem and it is the reason why I did the whole "ISO-8601 with UTC only" stuff (also because it is used like this in other modules). It can/will lead to issues when comparing timestamps with ">=" and so on.